### PR TITLE
XSS Sanitization in Operations modal

### PR DESCRIPTION
**Pull request by garunagiri**
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Tested manually to ensure both existing functionality is not broken and that the vulnerability is mitigated.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: templates/operations.html [hunk 1]
```diff
@@ -460,7 +464,7 @@ <h1 id="time-tactic" class="member-title"></h1>
         function operationCallback(data){
             function spacing() { return "&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;" }
             let OPERATION = data[0];
-            $("#op-control-state").html(OPERATION.state + spacing() + OPERATION.start + spacing() + OPERATION.chain.length + ' decisions');
+            $("#op-control-state").html(encodeHTML(OPERATION.state) + spacing() + encodeHTML(OPERATION.start) + spacing() + OPERATION.chain.length + ' decisions');
```
[the remark this answers is not in the document]
Comment (garunagiri) [reply templates/operations.html:467]: Doing `.text(OPERATION.state + "    |    " + OPERATION.start + "    |    " + OPERATION.chain.length + ' decisions')`
results in 
image


Using `.html()` instead makes it look like:
image

If we want to keep the formatting the same way, I think we might have to stick with `.html()` with `encodeHTMK()`

[the remark this answers is not in the document]
Comment (b17zr) [reply templates/operations.html:467]: Would adding multiple spaces around the pipes work?

[the remark this answers is not in the document]
Comment (garunagiri) [reply templates/operations.html:467]: I didn't even realize they didn't show up. I actually did have multiple spaces in my code:
image

I think GitHub took them out when I pasted the code here.

[the remark this answers is not in the document]
Comment (b17zr) [reply templates/operations.html:467]: And the non-breaking spaces don't work with .text() ?

[the remark this answers is not in the document]
Comment (garunagiri) [reply templates/operations.html:467]: No, the ampersands are escaped, so it shows up just as '&nbsp;' on the page

[the remark this answers is not in the document]
Comment (b17zr) [reply templates/operations.html:467]: Something like this should work:

```
$('#op-control-state').text(OPERATION.state + '|' + OPERATION.start + '|' + OPERATION.chain.length + ' decisions').html(function () {
  return $(this).html().replaceAll('|', '&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;')
}));
```

**Review by b17zr — APPROVED** [review]

**Review by protoli — APPROVED** [review]

**Pull request merged**